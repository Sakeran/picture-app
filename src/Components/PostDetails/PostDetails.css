.PostDetails {
  width: 75%;
  padding: 1em;
  background: white;
}

.PostDetails-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.PostDetails-creator-info {
  margin-top: 0.5em;
  font-size: 0.9em;
  font-weight: lighter;
}

.PostDetails-creator-info a {
  color: #6d1cba;
  text-decoration: none;
}

.PostDetails-description {
  max-width: 75%;
  margin: 0 auto;
  text-align: center;
}
.PostDetails-description p {
  display: inline-block;
  text-align: left;
}
