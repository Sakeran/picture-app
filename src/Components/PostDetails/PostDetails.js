import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './PostDetails.css';

const PostDetails = ({title, description, postDate, creator: {username, id}}) => (
  <div className="PostDetails section-border">
    <div className="PostDetails-info">
      <h2>{title}</h2>
      <span className="PostDetails-creator-info">
        Posted by <Link to={`/profile/${id}`}>{username}</Link> on {postDate}
      </span>
    </div>
    <div className="PostDetails-description">
      <p>{description}</p>
    </div>
  </div>
);

PostDetails.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  postDate: PropTypes.string.isRequired,
  creator: PropTypes.object.isRequired
}

export default PostDetails;
