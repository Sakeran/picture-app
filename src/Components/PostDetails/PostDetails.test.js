import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PostDetails from './PostDetails';

test('PostDetails component renders correctly', () => {
    const props = {
      title: 'Test Post',
      description: 'A test post.',
      postDate: 'January 1st 2001',
      creator: {
        id: '1111',
        username: 'Test User'
      }
    };
    const component = shallow(<PostDetails {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
});
