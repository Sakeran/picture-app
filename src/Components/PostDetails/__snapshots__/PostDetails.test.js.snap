// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostDetails component renders correctly 1`] = `
<div
  className="PostDetails"
>
  <div
    className="PostDetails-info"
  >
    <h2>
      Test Post
    </h2>
    <span
      className="PostDetails-creator-info"
    >
      Posted by 
      <Link
        replace={false}
        to="/profile/1111"
      >
        Test User
      </Link>
       on 
      January 1st 2001
    </span>
  </div>
  <div
    className="PostDetails-description"
  >
    <p>
      A test post.
    </p>
  </div>
</div>
`;
